usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlock.java

```diff
@@ -0,0 +1,264 @@
+                final Integer currentTripFirstTime = currentTripFirstStopTimeInSequence.getArrivalTime();
+                final Integer currentTripLastTime = currentTripLastStopTimeInSequence.getDepartureTime();
+
+                visitedTripIdCollection.add(currentTripId);
+
+                if (currentTripFirstTime != null && currentTripLastTime != null) {
+                    if (currentTripFirstTime <= currentTripLastTime) {
+                        tripCollectionFromSameBlockId.forEach(unvisitedTrip -> {
+                            final String unvisitedTripId = unvisitedTrip.getTripId();
+                            if (!visitedTripIdCollection.contains(unvisitedTripId)) {
+                                // if trips have the same service_id
+                                // | routeId | tripId | serviceId | blockId | first stop time | last stop time |
+                                // |---------|--------|-----------|---------|-----------------|----------------|
+                                // | 0       | 2      | a         | 5       | 07:00           | 10:00          |
+                                // | 0       | 5      | a         | 5       | 08:45           | 11:20          |
+                                // | 0       | 8      | a         | 5       | 11:40           | 13:50          |
+
+                                // fetch related StopTime for unvisitedTrip
+                                final SortedMap<Integer, StopTime> stopTimesForUnvisitedTrip =
+                                        dataRepo.getStopTimeByTripId(unvisitedTripId);
+
+                                final StopTime unvisitedTripFirstStopTimeInSequence =
+                                        stopTimesForUnvisitedTrip.get(stopTimesForUnvisitedTrip.firstKey());
+                                final StopTime unvisitedTripLastStopTimeInSequence =
+                                        stopTimesForUnvisitedTrip.get(stopTimesForUnvisitedTrip.lastKey());
+
+                                final Integer unvisitedTripFirstTime =
+                                        unvisitedTripFirstStopTimeInSequence.getArrivalTime();
+                                final Integer unvisitedTripLastTime =
+                                        unvisitedTripLastStopTimeInSequence.getDepartureTime();
+
+                                if (unvisitedTrip.getServiceId().equals(currentTrip.getServiceId())) {
+
+                                    // stop_times arrival_time and departure_time should not be null to perform the
+                                    // subsequent operations
+                                    if (unvisitedTripFirstTime != null && unvisitedTripLastTime != null) {
+                                        // stop_times arrival_time and departure_time are supposed to be ordered
+                                        if (unvisitedTripFirstTime <= unvisitedTripLastTime) {
```

Comment:
The trip first and last time can't be equal, right? Otherwise the trip is `0` seconds long?

Also, if this is another sanity check, I'd suggest refactoring to `areTripTimesValid()` method to better explain what's happening.

Reply:
Indeed, these cannot be equal. Will modify.